File: Compare-Version-Numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        vector<string>v1;
        vector<string>v2;
        stringstream ss1(version1);
        string token1;
        while (getline(ss1, token1, '.')) {
            v1.push_back(token1);
        }
        stringstream ss2(version2);
        string token2;
        while(getline(ss2, token2, '.')){
            v2.push_back(token2);
        }

        int maxLen = max(v1.size(), v2.size());
        for(int i=0; i<maxLen; i++){
            int r1 , r2;
            if(i < v1.size()){
                r1 = stoi(v1[i]);
            }else {
                r1 = 0;
            }
            if(i < v2.size()){
                r2 = stoi(v2[i]);
            }else {
                r2 = 0;
            }
            if(r1 > r2) return 1;
            else if(r1 < r2) return -1;
        }
        return 0;
    }
};

File: Compare-Version-Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        revision1 = version1.split('.')
        revision2 = version2.split('.')       
        len1, len2 = len(revision1), len(revision2)
        max_len = max(len1, len2)

        for i in range(max_len):
            r1 = int(revision1[i]) if i < len1 else 0
            r2 = int(revision2[i]) if i < len2 else 0

            if r1 < r2 :
                return -1
            elif r1 > r2 :
                return 1

        return 0